{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "UnboundLocalError",
     "evalue": "local variable 'count' referenced before assignment",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mUnboundLocalError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-1-55f193ff4762>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     53\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     54\u001b[0m \u001b[1;32mif\u001b[0m \u001b[0m__name__\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;34m\"__main__\"\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 55\u001b[1;33m     \u001b[0mfafile2dict\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m<ipython-input-1-55f193ff4762>\u001b[0m in \u001b[0;36mfafile2dict\u001b[1;34m()\u001b[0m\n\u001b[0;32m     45\u001b[0m     \u001b[1;31m# print(count)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     46\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 47\u001b[1;33m     \u001b[0mplt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbar\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mrange\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcount\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcount\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     48\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     49\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mUnboundLocalError\u001b[0m: local variable 'count' referenced before assignment"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "\n",
    "def fafile2dict():\n",
    "    '''\n",
    "    read a single FASTA file (SHH.fa) into a dictionary object\n",
    "    and print out the dictionary\n",
    "\n",
    "    Rerurn\n",
    "    --------------\n",
    "    seq:dict\n",
    "    the single gene dict, {descriptor_str:sequence_str}\n",
    "    --------------\n",
    "    '''\n",
    "    line = sys.stdin.readline().replace('\\n','')\n",
    "    seq = {}\n",
    "    while line != '':\n",
    "        if line[0] == '>':\n",
    "            name = line.replace('\\n','')\n",
    "            seq[name] = ''\n",
    "        else:\n",
    "            seq[name] += line.replace('\\n','').strip()\n",
    "        line = sys.stdin.readline()\n",
    "    print(seq)\n",
    "    return seq\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    fafile2dict()\n",
    "\n",
    "\n",
    "\n",
    "def fafile2dict():\n",
    "    '''\n",
    "    this function can be used to calculate the As, Cs, Gs, Ts in entire genome\n",
    "    \n",
    "    STDIN:\n",
    "    ----------------------------------------\n",
    "    the fasta file\n",
    "    run as 'python3 ques1.py yeast.fa'\n",
    "    ----------------------------------------\n",
    "\n",
    "    Return:\n",
    "    ----------------------------------------\n",
    "    base_a:int\n",
    "    the number of As\n",
    "    base_c:int\n",
    "    the number of Cs\n",
    "    base_g:int\n",
    "    the number of Gs\n",
    "    base_t:int\n",
    "    the number of Ts\n",
    "    ----------------------------------------\n",
    "    '''\n",
    "    line = sys.stdin.readline().replace('\\n','')\n",
    "    seq = {}\n",
    "    while line != '':\n",
    "        if line[0] == '>':\n",
    "            name = line.replace('\\n','')\n",
    "            seq[name] = ''\n",
    "        else:\n",
    "            seq[name] += line.replace('\\n','').strip()\n",
    "        line = sys.stdin.readline()\n",
    "    for bp in seq.values():\n",
    "        bp_list = list(bp)\n",
    "        count = []\n",
    "        for i in range(int(len(bp_list)/100)):\n",
    "            bp_frag = bp_list[(100*i-100):100*i]\n",
    "            base_gc = 0\n",
    "            for bp_sort in bp_frag:\n",
    "                if bp_sort == 'G' or bp_sort == 'C':\n",
    "                    base_gc += 1\n",
    "            count.append(base_gc)\n",
    "    # print(count)\n",
    "\n",
    "    plt.bar(range(len(count)), count)\n",
    "\n",
    "\n",
    "\n",
    "    # print('A:',base_a,'T:',base_t,'C:',base_c,'G:',base_g) \n",
    "    # return base_a, base_t, base_c, base_g\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    fafile2dict()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.7.4 64-bit ('base': conda)",
   "language": "python",
   "name": "python37464bitbaseconda7488c6938816401994c2e6b64e34356c"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
